src/app/pages/swap/components/swap-amount-field.tsx

```diff
@@ -0,0 +1,56 @@
+import { ChangeEvent } from 'react';
+
+import { Input, Stack, color } from '@stacks/ui';
+import { useField, useFormikContext } from 'formik';
+
+import { useShowFieldError } from '@app/common/form-utils';
+import { Caption } from '@app/components/typography';
+
+import { SwapFormValues } from '../hooks/use-swap';
+import { tempExchangeRate } from '../hooks/use-swappable-assets';
+
+interface SwapAmountFieldProps {
+  amountAsFiat: string;
+  isDisabled?: boolean;
+  name: string;
+}
+export function SwapAmountField({ amountAsFiat, isDisabled, name }: SwapAmountFieldProps) {
+  const { setFieldValue } = useFormikContext<SwapFormValues>();
+  const [field] = useField(name);
+  const showError = useShowFieldError(name);
+
+  async function onChange(event: ChangeEvent<HTMLInputElement>) {
+    const value = event.currentTarget.value;
+
+    // TODO: Need exchange rate here from asset
+    await setFieldValue('swapAmountTo', Number(value) * tempExchangeRate);
+
+    field.onChange(event);
+  }
+
+  return (
+    <Stack alignItems="flex-end" spacing="extra-tight">
+      <Input
```

Comment:
Inputs should have `label` elements with them. This helps make sure there's a large clickable area around the input. Clicking anywhere within the box that isn't the change asset dropdown, should focus the input.

Reply:
Also for accessibility.